Posts/2021/03/ScriptingGuy.0007.folderexists.md

```diff
@@ -0,0 +1,41 @@
+---
+post_title: Determine if a folder exists
+username: tfl@psp.co.uk
+Catagories: PowerShell
+tags: File, Test-Path
+Summary: How can I determine if a folder exists?
+---
+
+**Q:** Is there any way to determine whether or not a specific folder exists on a computer?
+**A:**  There are loads of ways you can do this.
+A simple way is to use the power of array handling in PowerShell.
+
+## The Test-Path Cmdlet
+
+PowerShell has a handy cmdlet for that, called `Test-Path`.
+It looks for a given path and returns `True` if it exists, otherwise it returns `False`.
+You could evaluate the result of the `Test-Path` like in the code snippet below 
+
+```powershell
+$Folder = 'C:\Windows'
+"Test to see if folder [$Folder]  exists"
+if (Test-Path -Path $Folder) {
+    "Path exists!"
+} else {
+    "Path doesn't exist."
+}
+```
+This is similar to the `-d` parameter in Bash.
```

Comment:
which command does -D refer to?

Reply:
It's used in Bash IF statements. Resolved.